analysis/topEFT/topeft.py

```diff
@@ -51,7 +51,7 @@ def construct_cat_name(chan_str,njet_str=None,flav_str=None):
 
 class AnalysisProcessor(processor.ProcessorABC):
 
-    def __init__(self, samples, wc_names_lst=[], hist_lst=None, do_errors=False, do_systematics=False, split_by_lepton_flavor=False, skip_signal_regions=False, skip_control_regions=False, dtype=np.float32):
+    def __init__(self, samples, wc_names_lst=[], hist_lst=None, do_errors=False, do_systematics=False, split_by_lepton_flavor=False, skip_signal_regions=False, skip_control_regions=False, dtype=np.float32, muonSyst='nominal'):
```

Comment:
Not a huge deal, but I personally prefer `dtype` to be the last argument.

Reply:
@andrea21z, I am not sure if I understand why we need to pass this argument to the processor? For the other systematics, we either calculate the up/down if the `--do-systs` flag is on, or we skip the up/down if that flag is off. I am wondering if there is a reason why we'd want to handle this systematic in a special way (i.e. be able to turn it on or off separately from the others)?

Reply:
Well, this uncertainty affects the muon selection, so has to be propagated to the jet selection and all the categories (it's not as simple as changing a weight). The same for the jet energy scale and resolution uncertainties. So one option is re-running the full analysis changing the pT of the muons , as in Andrea's version (there are other analysis with coffea that do this). The other option would be adding tags for the muon pT to basically everything in the analysis (categories, number of jets, HT...) and fill the nominal/up/down axis depending on the chosen muon pT. A third option would be implementing a 'for' loop at the beginning of the analysis, before the muon selection, and remove from the loop those elements that are not affected by muon pT (for example, the electron selection).

Reply:
Thanks for explaining this @GonzalezFJR. I think it makes sense to merge the PR with the way it's currently implemented. However, I think the third option you described above sounds like what we would want for the long term solution. We could then handle the other similar systematics (jet energy corrections, and maybe others?) as part of the same loop. That way, we could run everything all together in one run of the processor, instead of having to run separately for some of the systematics. I am wondering what your thoughts would be about this? If @GonzalezFJR and @bryates and @andrea21z agree that option 3 is the way we want to handle this, then I'm wondering if this is something @andrea21z you would be interested in working on?

However, I should reiterate that I would be in favor of merging this PR as it currently stands (and updating the handling of those systematics in a later branch). I can make an issue for the discussion of the potential updates to the way these systematics are handled, so that we can move the discussion there, since we'll be merging this PR soon, so it would not be very easy to continue the discussion here.

Reply:
Here is the issue: #123.

Reply:
I'm in favor of merging now, and following up with the issue later.

Reply:
Sounds good. I can start to work on the implementation of the loop if you think that is the best option.